server/tests/integration/listings.test.ts

```diff
@@ -0,0 +1,72 @@
+/**
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import request from 'supertest';
+
+import app from '../../src';
+import listingsFixtures from '../fixtures/listings';
+import merchantFixtures from '../fixtures/merchants';
+
+describe('Listings endpoints', () => {
+  describe('GET /listings', () => {
+    test('Should fetch all listings', async () => {
+      const expectedListingData = listingsFixtures.responseData;
+
+      const res = await request(app).get('/listings');
+
+      expect(res.status).toBe(200);
+      expect(res.body).toMatchObject(expectedListingData);
+    });
+
+    test('Should filter by listing ids', async () => {
```

Comment:
we should split this into two tests: one with single id, another one with multiple ids.
we should try to make tests simple, and avoid extra logic (e.g. map/forEach here).